File: leetcode/practice/202-happy-number/sol.cpp
class Solution {
public:
    bool isHappy(int n) ;
};

bool Solution :: isHappy(int n)
{
    bool happy[10] = {false , true , false , false , false , false , false , true , false , false} ;
    int temp = n ;
    while(n != 1)
    {
        int num = 0 ;
        while(n != 0)
        {
            int remainder = n%10 ;
            num += remainder*remainder ;
            n /= 10 ;
        }
        n = num ;
        if(n < 10 and !happy[n])
            return false ;
    }
    return true ;
}

File: leetcode/practice/202-happy-number/sol.py
class Solution:
    def isHappy(self, n: int) -> bool:
        happy = [False , True , False , False , False , False , False , True , False , False]
        while(n != 1) :
            num = 0
            while n != 0 :
                remainder = n%10
                n = n//10
                num += remainder**2
            n = num
            if n < 10 and happy[n] == False :
                return False
        return True
